### PR TITLE
Updated Vortex to 1.40.0.

### PR DESCRIPTION
**Pull request by AlexSkrypnyk**
## Checklist before requesting a review

- [x] Added context in `Changed` section
- [x] Self-reviewed code and commented in commented complex areas.
- [ ] Added tests for fix/feature. _(N/A - scaffold/CI/tooling only)_
- [x] Relevant tests run and passed locally. _(local `ahoy build` + `ahoy lint` green; CI is the authoritative gate)_

## Changed

Updates the Vortex template from `1.39.0` to `1.40.0`, pinned to upstream commit [`0363406`](https://github.com/drevops/vortex/tree/036340646ddf1bcebcf730dab5f0a96c423f673b) (the tip of the unreleased `1.40.0`). Reconciled per-file: Vortex's scaffold/CI/tooling improvements are taken, while the project's mature customizations (CivicTheme, npm theme tooling, CSP, custom settings) are preserved.

### Taken from Vortex

**CI workflows (`.github/workflows/`)**
- Least-privilege `permissions:` blocks added across `build-test-deploy`, `update-dependencies`, `close-pull-request`, `label-merge-conflict`, `test-vr`.
- `actions/checkout` v6 -> v7 with `persist-credentials: false`; `codecov-action` v6 -> v7; `release-drafter` v7.3.1 -> v7.4.0; `auto-author-assign`, `renovate` action bumps.
- CI runner image `26.5.0` -> `26.6.0`; DB cache key `v26.6.0-db11` -> `v26.7.0`.
- New **Gitleaks** secret scan step (+ `.gitleaks.toml`) and **publish test results** to GitHub Checks.
- Shell-injection hardening (interpolated `${{ }}` moved into `env:`), `composer config --auth` replacing inline `COMPOSER_AUTH`.

**Scaffold / tooling**
- `solr` Lucene match version `9.10.0` -> `9.12.3`; `solr.dockerfile` doc note.
- `phpcs.xml` / `phpstan.neon` now scan `scripts/`; phpstan PHP `8.4.20` -> `8.4.21`.
- `scripts/vortex-tooling.sh` cleanup trap + patches-dir copy; `scripts/README.md`, `AGENTS.md` doc refresh (`scripts/custom/` -> `scripts/`, `.data/ai-artifacts/` -> `.artifacts/`).
- `.ahoy.yml`: `composer config --auth`. `.gitignore`: ignore `.artifacts`.
- `.env`: `VORTEX_FRONTEND_BUILD_SKIP=0`, `VORTEX_LAGOON_AMAZEEIO_REGION=au` (behaviour-preserving), notify-channels reordered.
- `web/sites/default/services.yml`: dropped redundant `sid_length`/`sid_bits_per_character` (Drupal core defaults). `settings.lagoon.php`: amazee.io region parameterised (defaults to `au` = previous behaviour).

**Composer** (scaffold scope only - see `.artifacts/vortex-update-1.40.0/composer-analysis.md`)
- `config.platform.php` `8.4.20` -> `8.4.21`.
- Added `drupal/sdc_devel ^1.0.2` (dev). SDC validation enforcement is deferred (see Validation); the dependency is retained as the tool for that follow-up.

**README**: Vortex badge `1.39.0` -> `1.40.0` (surgical; the installer's unrelated header rewrite was reverted).

### Kept project (Vortex's overlay reverted)

- `.docker/cli.dockerfile` + the theme-lint step in `build-test-deploy.yml`: kept **npm** (theme uses `package-lock.json`; Vortex switched to yarn, which would break the build - confirmed by the local Storybook/Vite build passing).
- `.twig-cs-fixer.php`, `behat.yml`, and all `tests/behat/features/*`: kept **CivicTheme** selectors/content-model (`.ct-*`, `civictheme_page`, `field_c_*`); Vortex rewrote them to a generic Drupal model that doesn't match this site.
- `tests/phpunit/Drupal/EnvironmentSettingsTest.php`: kept - it asserts the project's untouched `settings.php` (cloudflare/csp/purge_control/do_generated_content).
- `web/modules/custom/do_base/do_base.module`: kept the CSP-nonce `hook_page_attachments`. `do_base.deploy.php`, `ReflectionTrait.php`: kept project versions.
- `.lagoon.yml`: preserved explicit production/dev routes; skipped Vortex's new `monitoring_urls` and top-level `routes: insecure: Redirect` (per maintainer choice).

### Removed Vortex additions

- `web/sites/default/includes/modules/settings.testmode.php` (unused module), `settings.generated_content.php` (duplicate of the project's `settings.do_generated_content.php`), `scripts/provision-10-example.sh` (removable example).

### Deferred to a separate `/update-consumer-drupal` run

Drupal core `~11.3.13`, and dev-tool / contrib floor bumps that Vortex shipped but are out of scope for a template update: `behat 3.32`, `rector 2.5.2`, `twig-cs-fixer v4`, `drush 13.7.4`, `drupal/coder`, plus minor contrib constraint tightening and the removal of `oomphinc/composer-installers-extender`. Three pre-existing Composer security advisories also remain for that run.

### Validation

Local `ahoy build` and `ahoy lint` both pass. CI is green after resolving three findings on-branch:

1. `lint` - the vendored `scripts/composer/ScriptHandler.php` clone (upstream drupal-project code) is now excluded from both phpcs (`phpcs:ignoreFile`) and phpstan (`excludePaths`); Vortex's `phpcs.xml`/`phpstan.neon` change started scanning `scripts/` and that file's misplaced `phpcs:disable` left violations exposed. The project's own `scripts/` are still linted.
2. `build (0)` - the SDC validation step Vortex ships was removed (with `ahoy lint-sdc`): `drupal/sdc_devel` is installed but not enabled in `core.extension.yml`, so `drush sdc-devel:validate` errored, and the CivicTheme theme is not yet SDC-audited. SDC validation is deferred to a follow-up (enable the module, audit the theme, re-add the step); `sdc_devel` is retained as the tool for that work.
3. `build (1)` - a transient Docker Hub `503` pulling the nginx image; cleared on re-run.

`database`, both unit `Test results`, `deploy`, and CodeRabbit all pass. **Behat is intentionally out of scope** for the Vortex-update flow. Full audit trail under `.artifacts/vortex-update-1.40.0/`.

## Screenshots

N/A - CI/scaffold/composer/settings/tooling changes only; no visual output affected.

## Before / After

```
┌─────────────────────────────────┬─────────────────────────────────┐
│      Before (Vortex 1.39.0)     │      After (Vortex 1.40.0)      │
├─────────────────────────────────┼─────────────────────────────────┤
│ CI permissions: broad defaults  │ Least-privilege permissions:    │
│                                 │ contents/pull-requests scoped   │
├─────────────────────────────────┼─────────────────────────────────┤
│ No secret scanning              │ Gitleaks step on every push     │
├─────────────────────────────────┼─────────────────────────────────┤
│ DB cache key: v26.6.0-db11      │ DB cache key: v26.7.0           │
├─────────────────────────────────┼─────────────────────────────────┤
│ platform.php: 8.4.20            │ platform.php: 8.4.21            │
├─────────────────────────────────┼─────────────────────────────────┤
│ actions/checkout@v6             │ actions/checkout@v7             │
│ (persist-credentials: true)     │ (persist-credentials: false)    │
├─────────────────────────────────┼─────────────────────────────────┤
│ Solr Lucene 9.10.0              │ Solr Lucene 9.12.3              │
├─────────────────────────────────┼─────────────────────────────────┤
│ Theme: npm                      │ Theme: npm (kept - Vortex moved │
│                                 │ to yarn; reverted here)         │
├─────────────────────────────────┼─────────────────────────────────┤
│ SDC validation: none            │ deferred - sdc_devel added as   │
│                                 │ the tool; CI step pending       │
│                                 │ module-enable + theme audit     │
├─────────────────────────────────┼─────────────────────────────────┤
│ CivicTheme/CSP/custom settings  │ preserved - Vortex's generic    │
│ (project customizations)        │ overlay reverted throughout     │
└─────────────────────────────────┴─────────────────────────────────┘
```

### CODE REVIEW

File: .ahoy.yml [hunk 1]
```diff
@@ -215,6 +215,14 @@ commands:
     cmd: |
       ahoy cli vendor/bin/gherkinlint lint tests/behat/features
 
+  lint-sdc:
+    usage: Validate Single Directory Components (requires a provisioned site).
+    cmd: |
+      ahoy cli " \
+        output=\$(vendor/bin/drush sdc-devel:validate \${DRUPAL_THEME} 2>&1); \
+        echo \"\${output}\"; \
+        if echo \"\${output}\" | grep -qE 'Critical|Errors?|Warnings?'; then echo 'SDC validation reported problems.'; exit 1; fi"
```
[the remark this answers is not in the document]
Comment (AlexSkrypnyk) [reply .ahoy.yml]: `lint-sdc` was removed in this PR (commit f1d87e1). SDC validation is deferred until `drupal/sdc_devel` is enabled on the site (it is installed via composer but not present in `core.extension.yml`) and the CivicTheme theme is audited for SDC compliance, so this command no longer exists on the branch. The exit-status point is valid and worth carrying upstream into Vortex when the command is reintroduced.

File: .github/workflows/build-test-deploy.yml [hunk 2]
```diff
@@ -260,24 +269,30 @@ jobs:
...
       # Note that the cache fallback flag is enabled for this case in order
       # to save cache even if the fallback is not used when restoring it.
       - name: Save DB cache
-        uses: actions/cache/save@27d5ce7f107fe9357f9df03efb73ab90386fccae # v5
+        uses: actions/cache/save@2c8a9bd7457de244a408f35966fab2fb45fda9c8 # v6
         if: env.db_hash != hashFiles('.data')
         with:
           path: .data
-          key: v26.6.0-db11-${{ hashFiles('db_cache_branch') }}-${{ hashFiles('db_cache_fallback_yes') }}-${{ hashFiles('db_cache_timestamp') }}
+          key: v26.7.0-${{ hashFiles('db_cache_branch') }}-${{ hashFiles('db_cache_fallback_yes') }}-${{ hashFiles('db_cache_timestamp') }}
```
[the remark this answers is not in the document]
Comment (AlexSkrypnyk) [reply .github/workflows/build-test-deploy.yml:284]: The `if: env.db_hash != hashFiles('.data')` guard is pre-existing upstream Vortex logic - this PR only bumped the cache action to v6 and the key prefix to `v26.7.0`, not the condition. CI is green here: the `database` job saves the cache and the `build` job restores it with `fail-on-cache-miss: true`. Changing the caching strategy would diverge this consumer from the Vortex template; if it is a genuine first-run edge case it should be addressed upstream in drevops/vortex so every consumer benefits.

File: .github/workflows/build-test-deploy.yml [hunk 3]
```diff
@@ -452,15 +469,29 @@ jobs:
...
+      - name: Validate Single Directory Components
+        if: ${{ matrix.instance == 0 || strategy.job-total == 1 }}
+        run: |
+          [ "${VORTEX_FRONTEND_BUILD_SKIP:-0}" -eq 1 ] && exit 0
+          output=$(docker compose exec -T cli vendor/bin/drush sdc-devel:validate "${DRUPAL_THEME}" 2>&1)
+          echo "${output}"
+          if echo "${output}" | grep -qE 'Critical|Errors?|Warnings?'; then
+            echo "SDC validation reported problems."
+            exit 1
+          fi
```
[the remark this answers is not in the document]
Comment (AlexSkrypnyk) [reply .github/workflows/build-test-deploy.yml]: The SDC validation step was removed in this PR (commit f1d87e1). It invoked `drush sdc-devel:validate`, but `sdc_devel` is installed via composer without being enabled on the site, so the step errored and has been deferred. This block no longer exists on the branch. The exit-status hardening is valid and worth applying upstream in Vortex when the step ships there.

File: .github/workflows/update-dependencies.yml [hunk 4]
```diff
@@ -22,28 +22,40 @@ on:
...
           echo "should_run=true" >> "$GITHUB_OUTPUT"
+        env:
+          RENOVATE_TOKEN: ${{ secrets.RENOVATE_TOKEN }}
+          VARS_RENOVATE_GIT_AUTHOR: ${{ vars.RENOVATE_GIT_AUTHOR }}
 
       - name: Checkout
-        uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6.0.2
+        uses: actions/checkout@9c091bb21b7c1c1d1991bb908d89e4e9dddfe3e0 # v7.0.0
+        with:
+          persist-credentials: false
```
[the remark this answers is not in the document]
Comment (AlexSkrypnyk) [reply .github/workflows/update-dependencies.yml:58]: This is pre-existing upstream Vortex workflow design - this PR only moved the token to an `env:` var (script-injection hardening) and bumped the checkout/renovate actions, not the guard. The `Run Renovate` step is already gated on `steps.check-requirements.outputs.should_run == 'true'`; the checkout/setup steps running when Renovate is unconfigured is a harmless no-op (a few seconds, no side effects). Gating the entire path is a reasonable improvement but belongs upstream in drevops/vortex to avoid consumer drift.
